### DISCUSSION TITLE
jiffy clock location on c64

### DISCUSSION THREAD

**Discussion by xlar54** [cc65/cc65#2004] [General]
Anyone know if 0xa0-0xa2 are preserved for the kernal jiffy clock?

**Comment by mrdudz** [comment]
Thats correct, see eg here: http://unusedino.de/ec64/technical/aay/c64/zpmain.htm

**Reply by xlar54** [reply]
thanks.  Wasnt sure if cc65 took over those locations

**Comment by xlar54** [comment]
unless Im doing something wrong (which is probably the case), im getting strange readings on those locations.  Heres my sample code:

```
while(1)
    {
        jiffystart = ((*(uint8_t*) (0xa0)) * 10000) + ((*(uint8_t *) (0xa1)) * 100) + (*(uint8_t *) (0xa2));
        printf("%06ld\n", jiffystart);
    }
```

jiffystart is a uint32_t  value

Heres the output.  Notice the drop in value:

![image](https://user-images.githubusercontent.com/1909642/220763271-e029badc-23d4-4e63-bf82-22b4b57938a3.png)


is this a bug or am I doing something stupid?

**Comment by mrdudz** [comment]
multiply by 0x100 and 0x10000 :)

**Reply by xlar54** [reply]
yeah im doing that 

`jiffystart = ((*(uint8_t*) (0xa0)) * 10000) + ((*(uint8_t *) (0xa1)) * 100) + (*(uint8_t *) (0xa2));`

**Reply by oqibidipo** [reply]
No, you aren't. You are multiplying by decimal 10000 and 100, not hexadeximal 0x10000 and 0x100.

**Reply by xlar54** [reply]
hmm.. but im derefencing the values as integers, then multiplying by 10000 and 100.  or am i? lol

**Reply by xlar54** [reply]
yep- youre right.  dummy mistake.  thank you!!!
